### PR TITLE
🎉 New Source:  ClickUp

### PR DESCRIPTION
**Pull request by josix**
## What
Adding a source connector for ingesting data from the ClickUp by using low-code configuration. ([ref](https://github.com/airbytehq/connector-contest/issues/106))

## How
Following the instruction mentioned in [tutorials](https://docs.airbyte.com/connector-development/config-based/low-code-cdk-overview/) to update `clickup_api.yaml`, `spec.yaml`, `schemas/*.json`, and `configured_catalog.json`.

## Recommended reading order
1. `spec.yaml`
2. `clickup_api.yaml`
3. `schemas/*.json`
4. `configured_catalog.json`

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

image

</details>

### CODE REVIEW

**Comment by josix** [comment]
Hello team, I'm confused about the way to make the tests. I've passed the integration test by running `python -m pytest integration_tests -p integration_tests.acceptance` mentioned in the [low-code tutorial](https://docs.airbyte.com/connector-development/config-based/tutorial/testing), but I also found the instructions mentioned in [`README`](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source-configuration-based/README.md.hbs#L48) and one [`acceptance-test-docker.sh`](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/source_acceptance_test_files/acceptance-test-docker.sh) to run another acceptance test. It seems like they are running different stuff (plz correct me if I got this wrong). Does that mean I should run and pass all three methods to check whether this PR is okay? I'm unsure if there are existent documents or tutorials to help me understand how to conduct unit tests, integration tests, and acceptance tests for low-code connectors. Could you provide me with some resources about it? Thanks a lot.🙏🏻

File: airbyte-integrations/connectors/source-clickup-api/integration_tests/catalog.json [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{
+  "streams": [
+    {
+      "name": "TODO fix this file",
+      "supported_sync_modes": ["full_refresh", "incremental"],
+      "source_defined_cursor": true,
+      "default_cursor_field": "column1",
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-clickup-api/integration_tests/catalog.json:7]: Remove this file.

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/schemas/TODO.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# TODO: Define your stream schemas
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/schemas/TODO.md:1]: Remove this file.

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/spec.yaml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: ClickUp Api Spec
+  type: object
+  required:
+    - Authorization
+  additionalProperties: true
+  properties:
+    # 'TODO: This schema defines the configuration required for the source. This usually involves metadata such as database and/or authentication information.':
+    Authorization:
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/spec.yaml:11]:
```suggestion
    api_token:
```
Please rename to better naming.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
@josix do you need any further assistance here?

**Comment by josix** [comment]
Hi @marcosmarxm thank you for the reminder and reviewing, I'm still working on that. But I guess I will be off here for around 2-3 days since I'm working on my master's degree oral exam at the same time. I'll continue to create some endpoints for `sub-stream` and `incremental stream`. Do you have some documents or examples about these? I think I could have a look before I start to implement. That would be helpful. Thanks a lot.

**Comment by marcosmarxm** [comment]
@josix I lost the notification of your message. Do you still need assistance here?

**Comment by josix** [comment]
> @josix I lost the notification of your message. Do you still need assistance here?

Yes, please. I would start to continue the work this week although the hacktoberfest is gone lol.

**Comment by marcosmarxm** [comment]
> 

Because this was submitted before 2-november the PR is eligible to receive the prize.

**Comment by marcosmarxm** [comment]
@josix there are some tutorials here: https://docs.airbyte.com/connector-development/config-based/tutorial/getting-started besides that please point what is your question to help your, or enter our Slack and ask for help in `#hacktoberfest-2022` channel.

**Comment by marcosmarxm** [comment]
Hello! I'm going to be out of the office this Friday and won't be able to review your contribution again today, I return next Monday. So far, most contributions look solid and are almost done to be approved. As said in [Chris' comment](https://airbytehq.slack.com/archives/C0450PCSDB2/p1667430678866629) all contributions made before 2-November are eligible to receive the prize and have 2 weeks to merge the contributions. But I ensure next week we're going to have your contribution merged. If you have questions about the implementation you can send them in `#hacktoberfest-2022` in Airbyte's Slack.

Sorry the inconvenience and see you again next week, thank you so much for your contribution!

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/spec.yaml (hunk 3 above)
Comment (josix) [reply airbyte-integrations/connectors/source-clickup-api/source_clickup_api/spec.yaml]: Thanks for pointing this out. I've updated this.

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/schemas/TODO.md (hunk 2 above)
Comment (josix) [reply airbyte-integrations/connectors/source-clickup-api/source_clickup_api/schemas/TODO.md]: Removed at `8a2d8e7ef`

File: airbyte-integrations/connectors/source-clickup-api/integration_tests/catalog.json (hunk 1 above)
Comment (josix) [reply airbyte-integrations/connectors/source-clickup-api/integration_tests/catalog.json]: Removed at `8a2d8e7ef`

**Comment by josix** [comment]
Hello @marcosmarxm, thank you for letting me know. I've added docs and removed some redundant files following your comments. Please have a look, thank you!

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      url_base: "https://api.clickup.com/api/v2"
+    name: "{{ options['name'] }}"
+    primary_key: "{{ options['primary_key'] }}"
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
+
+streams:
+  - type: DeclarativeStream
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml:32]: @josix are you planning to add more streams? (Tasks, Lists, etc?)

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+  retriever:
+    type: SimpleRetriever
+    $options:
+      url_base: "https://api.clickup.com/api/v2"
+    name: "{{ options['name'] }}"
+    primary_key: "{{ options['primary_key'] }}"
+    record_selector:
+      $ref: "*ref(definitions.selector)"
+    paginator:
+      type: NoPagination
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml:29]: Please add Pagination: https://clickup.com/api/clickupreference/operation/GetTasks/

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Some considerations

File: airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml (hunk 4 above)
Comment (josix) [reply airbyte-integrations/connectors/source-clickup-api/source_clickup_api/clickup_api.yaml:32]: Yeah, I'll add Tasks stream these days.

**Comment by marcosmarxm** [comment]
@josix let me know if you need any assistance to finish the changes

**Comment by josix** [comment]
@marcosmarxm sorry for late reply, I've added list and task stream, please have a look thanks!

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**